symbolic/src/Data/Macaw/Symbolic/Memory.hs

```diff
@@ -242,25 +243,68 @@ newGlobalMemoryWith
  -> MC.Memory (MC.ArchAddrWidth arch)
  -- ^ The macaw memory
  -> m (CL.MemImpl sym, MemPtrTable sym (MC.ArchAddrWidth arch))
-newGlobalMemoryWith hooks proxy bak endian mmc mem = do
+newGlobalMemoryWith hooks proxy bak endian mmc mem =
+  newMergedGlobalMemoryWith hooks proxy bak endian mmc [mem]
+
+-- | A version of 'newGlobalMemoryWith' that takes a list of memories and
+-- merges them into a flat addresses space.
```

Comment:
Can you document the required precondition that the address spaces must not overlap?

Also, it is worth a note that we will need to add configuration options to `GlobalMemoryHooks` to support other strategies for merging memories.